### PR TITLE
fixed highlight on wrong parts

### PR DESCRIPTION
**Pull request by acaldas**

### CODE REVIEW

File: src/js/visual/configurator.js [hunk 1]
```diff
@@ -46,9 +46,13 @@ ripe.Configurator.prototype.init = function() {
     // highlight operation (considers only the default ones)
     this.partsList = [];
     this.owner.getConfig(function(config) {
-        var defaults = config.defaults;
         this.hiddenParts = config.hidden;
-        this.partsList = Object.keys(defaults);
```
Comment (joamag) [inline src/js/visual/configurator.js]: this was the issue as the order was wrong right ?

Comment (acaldas) [reply src/js/visual/configurator.js]: No, the parts list is in alphabetic order. The problem was that the defaults contains hidden parts like the patch. And on Regent's case it as ```metal_caps```which no longer exists.

Comment (joamag) [reply src/js/visual/configurator.js]: this should definitly not pass the ripe building process, I need to create a verification for this ...

**Comment by joamag** [comment]
Once this build is complete (if it succeeds) re-test this and see if the "fix" is still required https://jenkins-fast.platforme.com/job/3d-build/921/

**Comment by joamag** [comment]
@acaldas please re-test with this build https://jenkins-fast.platforme.com/job/3d-build/922/console

**Comment by acaldas** [comment]
@joamag It's working now

may-10-2018 10-00-02

**Comment by joamag** [comment]
Cool, closing the issue then

**Pull request closed without merging**